wildfly-users-web/src/main/java/gallb/wildfly/users/web/PublicationMB.java

```diff
@@ -96,8 +98,56 @@ public void setCurrentPublication(Publication currentPublication) {
 		return publicationList;
 	}
 
-	public void store(Publication p_value) {// store newspaper, magazine or book
+	/**
+	 * Insert new Book, Magazine or Newspaper Use currentAuthors and
+	 * currentPublisher
+	 * 
+	 * @param p_title-new
+	 *            title
+	 * @param p_nrOfCopies-number
+	 *            of copies left, and copies on stock
+	 */
+	public void store(String p_title, String p_nrOfCopies) {
+		if(p_title.isEmpty() || p_nrOfCopies.isEmpty()){
+			MessageService.warn("All field is requered");
+			return;
+		}
+		int nrOfCopies;
+		try{
+			nrOfCopies=Integer.parseInt(p_nrOfCopies);
+		}catch (NumberFormatException e) {
+			MessageService.warn("number of copies is not a valid Number");
+			return;
+		}
+		oLogger.info("-------------nrOfCopies:  " + p_nrOfCopies + " type:  " + type);
+		Publication p_value;
 		try {
+			switch (type) {

```

Comment:
csak siman `type ` helyett `this.type` -> jobban latszik hogy ez egy field (ez ervenyes mindenhol)